src/test/groovy/unit/org/pih/warehouse/core/StockMovementStatusSpec.groovy

```diff
@@ -0,0 +1,415 @@
+package unit.org.pih.warehouse.core
+
+import org.pih.warehouse.inventory.StockMovementStatusCode
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.order.OrderStatus
+import org.pih.warehouse.requisition.Requisition
+import org.pih.warehouse.shipping.Shipment
+import org.pih.warehouse.shipping.ShipmentStatus
+import org.pih.warehouse.shipping.ShipmentStatusCode
+import spock.lang.Specification
+import spock.lang.Unroll
+import org.pih.warehouse.requisition.RequisitionStatus
+import util.StockMovementContext
+import util.StockMovementStatusHelper
+
+@Unroll
+class StockMovementStatusSpec extends Specification {
+
+    void "should return shipment status #expected for inbound stock movement returns"() {
+        given:
+        StockMovementContext context = new StockMovementContext() {
+            @Override
+            boolean isReturn() {
+                return true
+            }
+            @Override
+            boolean isInbound() {
+                return true
+            }
+            @Override
+            boolean isOutbound() {
+                return false
+            }
+        }
+        when:
+        ShipmentStatusCode shipmentStatusCode = status
+        context.shipment = new Shipment() {
+            @Override
+            ShipmentStatus getStatus() {
+                return new ShipmentStatus(code: shipmentStatusCode)
+            }
+        }
+
+        then:
+        StockMovementStatusHelper.getStatus(context) == expected
+
+        where:
+        status || expected
+        ShipmentStatusCode.PENDING              ||  ShipmentStatusCode.PENDING
+        ShipmentStatusCode.SHIPPED              ||  ShipmentStatusCode.SHIPPED
+        ShipmentStatusCode.PARTIALLY_RECEIVED   ||  ShipmentStatusCode.PARTIALLY_RECEIVED
+        ShipmentStatusCode.RECEIVED             ||  ShipmentStatusCode.RECEIVED
+    }
+
+    void "should return requisition status #expected for outbound stock movement returns"() {
+        given:
+        StockMovementContext context = new StockMovementContext() {
```

Comment:
I recommend doing mocks/stubs this way instead of hard overriding the methods. It feels less hacky too me and it gives you the benefit of being able to do asserts on the stub such as asserting a specific method was called X times with Y args. Plus this way you can re-stub it later in the test if you want to change the behaviour.  You're not doing either here, but still I think it's a good practice to stub it out
```
StockMovementContext context = Stub(StockMovementContext) {
        isReturn() >> true
        isInbound() >> false
        isOutbound() >> true
}
```